bleak/__init__.py

```diff
@@ -758,8 +758,8 @@ async def write_gatt_char(
             response:
                 If ``True``, a write-with-response operation will be used. If
                 ``False``, a write-without-response operation will be used.
-                If omitted or ``None``, the "best" operation will be used
-                based on the reported properties of the characteristic.
+                For legacy reasons, if omitted or ``None`` the "write with response" operation
```

Comment:
This is not entirely correct as some peripherals may incorrectly indicate that they don't support write with response even if they really do. This is why it was worded the way it was (and why we want to discourage not providing the arg - because it can be inconsistent between peripherals or change if a peripheral firmware "fixes" a missing property).

Reply:
Sorry, I do not get your point, how the original is more accurate than my fix. A few lines later it says `response = "write" in characteristic.properties` so it goes for response, if "write" is declared. In case this is wrongly declared I assume it falls back to "write NR" in the backend, which I meant by `if allowed by the characteristic`. Should I change to `if supported by the device` or do you just want to keep your formulation?
Your code, your call, just wanted to say that it lead me in the wrong direction (because the tip was not visible in the VScode hints.)

Reply:
The new text doesn't mention the "write" property. The new text is only true if the property is correctly implemented by the peripheral. But there are peripherals out in the world that don't correctly implement the property. So it is possible that the peripheral actually does support write with response even if the "write" property is not present. The old text made it clear that the actual behavior it depends on if the "write" property is present or not.